SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/PartitionManager.cpp

```diff
@@ -0,0 +1,557 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LoadBalancing/PartitionManager.h"
+#include "EngineClasses/SpatialVirtualWorkerTranslator.h"
+#include "LoadBalancing/PartitionManager.h"
+#include "Schema/ServerWorker.h"
+#include "Schema/StandardLibrary.h"
+#include "SpatialView/SpatialOSWorker.h"
+#include "Utils/ComponentFactory.h"
+#include "Utils/InterestFactory.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialPartitionManager);
+
+namespace SpatialGDK
+{
+struct FPartitionInternalState
+{
+	Worker_PartitionId Id = 0;
+	QueryConstraint LBConstraint;
+
+	TOptional<Worker_RequestId> CreationRequest;
+	TOptional<Worker_RequestId> AssignmentRequest;
+
+	FString DisplayName;
+	void* UserData = nullptr;
+	bool bAcked = false;
+	bool bInterestDirty = true;
+	bool bPartitionCreated = false;
+
+	FLBWorkerHandle UserAssignment;
+	FLBWorkerHandle RequestedAssignment;
+	FLBWorkerHandle CurrentAssignment;
+};
+
+struct FLBWorkerInternalState
+{
+	Worker_EntityId SystemWorkerId;
+	Worker_EntityId ServerWorkerId;
+
+	FName WorkerType;
+	FString FullWorkerName;
+};
+
+FLBWorkerDesc::FLBWorkerDesc(FName InWorkerType)
+	: WorkerType(InWorkerType)
+{
+}
+FLBWorkerDesc::~FLBWorkerDesc() = default;
+
+FPartitionDesc::~FPartitionDesc() = default;
+
+namespace
+{
+CommandRequest CreateClaimPartitionRequest(Worker_PartitionId Partition)
+{
+	Worker_CommandRequest ClaimRequestData = Worker::CreateClaimPartitionRequest(Partition);
+	return CommandRequest(OwningCommandRequestPtr(ClaimRequestData.schema_type), ClaimRequestData.component_id,
+						  ClaimRequestData.command_index);
+}
+} // namespace
+
+struct FPartitionManager::Impl
+{
+	Impl(ViewCoordinator& Coordinator, TUniquePtr<InterestFactory>&& InInterestF)
+		: WorkerView(Coordinator.CreateSubView(SpatialConstants::SERVER_WORKER_COMPONENT_ID, SpatialGDK::FSubView::NoFilter,
+											   SpatialGDK::FSubView::NoDispatcherCallbacks))
+		, SystemWorkerView(Coordinator.CreateSubView(SpatialConstants::WORKER_COMPONENT_ID, SpatialGDK::FSubView::NoFilter,
+													 SpatialGDK::FSubView::NoDispatcherCallbacks))
+		, PartitionView(Coordinator.CreateSubView(SpatialConstants::PARTITION_ACK_COMPONENT_ID, SpatialGDK::FSubView::NoFilter,
```

Comment:
Perhaps we should be using a more generic Partition component here, or even fold it into the existing PARTITION_SHADOW_COMPONENT_ID. I see currently the INITIAL_STRATEGY_PARTITION_ENTITY_ID is missing it though.

Reply:
Some renaming is needed alright. Not sure what it should be just yet though. I'd keep both separated for the time being. There is a chance we do not need the ACK component.
What is your point about INITIAL_STRATEGY_PARTITION_ENTITY_ID ? What is missing?

Reply:
`INITIAL_STRATEGY_PARTITION_ENTITY_ID` is missing the `PARTITION_SHADOW_COMPONENT`